addons/stock/migrations/11.0.1.1/post-migration.py

```diff
@@ -189,6 +195,64 @@ def create_stock_move_line(env):
     )
 
 
+@openupgrade.logging()
+def fill_missing_lots_for_sml(env):
+    """On v10, `reserved_quant_id` field on stock.move.operation.link is
+    filled unless manual operations or extra quantities are added on transfer
+    wizard, which technically means that they fall under this part of code:
+
+    https://github.com/OCA/OpenUpgrade/blob/10.0/addons/stock/models/
+    stock_picking.py#L759
+
+    The SQL query for inserting stock.move.line can't be modified as explained
+    in previous method, so we call this method afterwards for filling missing
+    lots when possible.
+
+    Heuristic:
+
+    * Find stock.move.operation.link records without reserved quant, but the
+      linked operation has some lots.
+    * Check the stock.move.line (sml) new records associated to the move. If
+      there's only one sml with the same quantity of stock.pack.operation.
+    * If found, then we can conclude the lot is that one.
+    * If not, we try to see that all the stock.move.line of the same product
+      makes the total of the stock.pack.operation.lot for concluding the same.
+    """
+    env.cr.execute("""
+        SELECT sm.id, spol.qty, spol.lot_id, spol.lot_name
+        FROM stock_pack_operation spo
+            JOIN stock_move_operation_link smol
+                ON smol.operation_id = spo.id
+            JOIN stock_move sm ON sm.id = smol.move_id
+            JOIN stock_pack_operation_lot spol
+                ON spol.operation_id = spo.id
+        WHERE smol.reserved_quant_id IS NULL""",
+    )
+    rounding = env['decimal.precision'].precision_get(
+        'Product Unit of Measure')
+    Move = env['stock.move']
```

Comment:
why not `move_model` instead?

Reply:
Well, for being shorter, and following latest Odoo convention about this (putting the name in upper).

Reply:
but OpenUpgrade should follow OCA convention, not Odoo convention 😁

Reply:
But there's no convention on that one :wink: